src/Sarif/Visitors/InsertOptionalDataVisitor.cs

```diff
@@ -253,6 +237,32 @@ public override Result VisitResult(Result node)
                 }
             }
 
+            if (_dataToInsert.HasFlag(OptionallyEmittedData.PartialFingerprints))
+            {
+                Location primaryLocation = node.Locations?.FirstOrDefault();
+                if (primaryLocation != null)
+                {
+                    PhysicalLocation physicalLocation = primaryLocation.PhysicalLocation;
+                    Uri resolvedUri = GetResolvedArtifactLocationUri(physicalLocation.ArtifactLocation);
+
+                    Region expandedRegion = _fileRegionsCache.PopulateTextRegionProperties(physicalLocation.Region, resolvedUri, false);
+                    Region contextRegion = _fileRegionsCache.ConstructMultilineContextSnippet(expandedRegion, resolvedUri);
+
+                    if (contextRegion?.Snippet?.Text != null)
+                    {
+                        string contextRegionHash = HashUtilities.ComputeStringSha256Hash(contextRegion.Snippet.Text);
+
+                        node.PartialFingerprints ??= new Dictionary<string, string>();
+
+                        if (!node.PartialFingerprints.ContainsKey(ContextRegionHash) ||
+                            _dataToInsert.HasFlag(OptionallyEmittedData.OverwriteExistingData))
```

Comment:
The opposite of these conditions can be placed at beginning so it can avoid unnecessary computing the region snippets and hashes.